Modules/_functoolsmodule.c

```diff
@@ -354,6 +491,8 @@ static PyMemberDef partial_memberlist[] = {
      "tuple of arguments to future partial calls"},
     {"keywords",        _Py_T_OBJECT,       OFF(kw),        Py_READONLY,
      "dictionary of keyword arguments to future partial calls"},
+    {"np",              Py_T_PYSSIZET,      OFF(np),        Py_READONLY,
```

Comment:
IMO, `np` is too vague for a public API. Suggestions:

- `numplaceholders`
- `num_placeholders`

Alternatively, make it "private" and prefix it with an underscore: `_numplaceholders`.

You also need to document it.